GetStream/stream-chat-android
# Feature/3240 support deleted message behavior customization

### 🎯 Goal

Fixes #3240 

### 🛠 Implementation details

Exposed a way to customize the deleted messages behavior in the MessageList. This can be customized by passing in the `DeletedMessagesVisibility` to the `MessageListViewModel`.

### 🎨 UI Changes

| NONE | OWN | ALL |
| --- | --- | --- |
| ![Screenshot_1649232027](https://user-images.githubusercontent.com/17215808/161927347-7ea8426b-2937-45fd-b83e-576c8244a3a5.png) | ![Screenshot_1649231833](https://user-images.githubusercontent.com/17215808/161927564-be7c0330-6049-4bf0-86b8-456cd9e7dfcf.png) | ![Screenshot_1649232045](https://user-images.githubusercontent.com/17215808/161927587-3432c93c-0fe5-443c-8c89-8a473828f0f6.png) |

### 🧪 Testing

1. Run the Compose Sample app and log in as Filip
2. Open the "Deleted Messages Test" channel and browse the messages.
3. By default the vbisibility is `OWN`, so you'll see just your own messages.
4. Open `MessagesActivity.kt` for Compose and change the following line:

```kotlin
        setContent {
            ChatTheme(
                dateFormatter = ChatApp.dateFormatter,
                deletedMessagesVisibility = DeletedMessagesVisibility.OWN // here
            ) {
                MessagesScreen(
                    channelId = channelId,
                    onBackPressed = { finish() },
                    onHeaderActionClick = {}
                )

                // MyCustomUi()
            }
        }
```

Change it to `NONE` or `ALL` and you should see the respective results above. `NONE` shows no deleted messages. `ALL` shows all of them (regardless of ownership). 

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

![delete](https://user-images.githubusercontent.com/17215808/161922603-fca0e443-a0e9-4282-95be-6806a60f7c0d.gif)
